core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -257,10 +262,16 @@ public void startUp(Session session, String keyspace, boolean resetDb, boolean c
                 .put(STRING, Functions::getStringDataPoint)
                 .build();
 
+        metricsIdxCache = CacheBuilder.newBuilder().expireAfterAccess(1, TimeUnit.HOURS).maximumSize(40000).build();
```

Comment:
I disagree about using a distributed cache. The purpose of this ticket is to speed requests for inserting data points. We can speed up requests by simply taking the index updates off the write path and doing them out of band in a background task. The cache is an additional optimization in that it reduces the amount and frequency of writes. I am very skeptical that the overhead of a distributed cache would be worth it.